bdk-ffi/src/types.rs

```diff
@@ -36,12 +39,29 @@ impl From<BdkCanonicalTx<'_, BdkTransaction, ConfirmationTimeHeightAnchor>> for
         };
 
         CanonicalTx {
-            transaction: Arc::new(Transaction::from(tx.tx_node.tx)),
+            transaction: Arc::new(Transaction::from(tx.tx_node.tx.as_ref().clone())),
             chain_position,
         }
     }
 }
 
+// impl From<BdkCanonicalTx<'_, BdkTransaction, ConfirmationTimeHeightAnchor>> for CanonicalTx {
```

Comment:
I think we can remove this commented out part here?

Reply:
Totally. My bad.